x/gamm/pool-models/balancer/pool.go

```diff
@@ -618,14 +618,23 @@ func (p Pool) SpotPrice(ctx sdk.Context, baseAsset, quoteAsset string) (sdk.Dec,
 		return sdk.Dec{}, errors.New("pool is misconfigured, got 0 weight")
 	}
 
+	defer func() {
+		// defer function to escape the panic when spot price overflows
+		if r := recover(); r != nil {
+			spotPrice = sdk.Dec{}
+			err = types.ErrSpotPriceOverflow
+		}
+	}()
+
 	// spot_price = (Base_supply / Weight_base) / (Quote_supply / Weight_quote)
 	// spot_price = (weight_quote / weight_base) * (base_supply / quote_supply)
 	invWeightRatio := quote.Weight.ToDec().Quo(base.Weight.ToDec())
 	supplyRatio := base.Token.Amount.ToDec().Quo(quote.Token.Amount.ToDec())
 	fullRatio := supplyRatio.Mul(invWeightRatio)
 	// we want to round this to `SigFigs` of precision
-	ratio := osmomath.SigFigRound(fullRatio, types.SigFigs)
-	return ratio, nil
+	spotPrice = osmomath.SigFigRound(fullRatio, types.SigFigs)
+
+	return spotPrice, err
```

Comment:
Needs to add an overflow check here right? (E.g. a maximum of 2**128 or 2**160)?

Reply:
Right! I just added a check that capped at 2^160, although we could technically increase it to 2^196 without overflowing on equal weight pools. I personally think the buffer is useful regardless but would be curious to hear your thoughts